### ISSUE TITLE
Checkout to O365 Office Online fails if OneDrive file already exists

### ISSUE THREAD

**Issue by gsteimer** [tsgrp/HPI#2356]
When checking out a file to O365 / Office Online, if the file somehow already exists on the user's OneDrive, the code fails with the following error:

```
Caused by: com.tsgrp.opencontent.core.exception.OCRuntimeException: Failed to checkout document(09000003800591d5) with status code: 200 and reason: OK
	at com.tsgrp.opencontent.core.action.executer.CheckoutToCloudServiceActionExecuter.executeImpl(CheckoutToCloudServiceActionExecuter.java:137) ~[oc-hpiDemoDctm.1.core-3.4.2.jar:3.4.2]
	... 50 more
```

You can recreate this issue by following these steps:

1. Checkout to OneDrive
2. Ensure the Cancel Checkout action is set up to NOT delete the cloud copy
3. Cancel the checkout
4. Attempt to checkout to OneDrive again (this error will occur)

It looks like it stems from this part of the code:

```java
// If the status code is 201 for successfully created then we do NOT want to be able to check the document out
int status = responseOfficeOnline.getStatusCode().value();
if(status != 201 ) {
	throw new OCRuntimeException("Failed to checkout document(" +  objectId + ") with status code: " + responseOfficeOnline.getStatusCode().value() + " and reason: " 
	+ responseOfficeOnline.getStatusCode().getReasonPhrase());
}
```

So in other words, unless the Graph API creates a document, our code will always fail.  In the case of the document already existing, I would think we should be smart enough to either:

1. open the existing file, ignoring the repo contents
2. overwrite the existing file with the version the user is trying to check out 
3. OR best option - give the user the ability to choose one of the above (this would be an enhancement).  If we choose one of the above options, then remove `Enhancement` here and open a new item.

**Comment by mwojno-tsg** [comment]
There is an upload (which the code is only using) and an update. We have to check and see if the document exists, if it does then to run an update instead to avoid this issue. 

https://docs.microsoft.com/en-us/graph/api/driveitem-put-content?view=graph-rest-1.0&tabs=http

**Comment by gsteimer** [comment]
Moved to Jira: https://alfresco.atlassian.net/browse/SEP-35
